shopinvader/models/shopinvader_partner.py

```diff
@@ -90,14 +90,24 @@ def _is_same_partner_value(self, partner, vals):
         """ we check if one of the given value is different than values
             of the given partner
         """
-        data = partner._convert_to_write(partner._cache)
-        for key in data:
-            if key not in vals:
+        skip_keys = self._is_same_partner_value_skip_keys(partner)
+        keys_to_check = []
+        for key in vals.keys():
+            if key in skip_keys or key not in partner:
                 continue
+            keys_to_check.append(key)
+            if key not in partner._cache:
```

Comment:
@simahawk We don't have to check if the key is already into the cache... it's done by the orm
https://github.com/odoo/odoo/blob/12.0/odoo/fields.py#L989